tests/server/sockfilt.c

```diff
@@ -544,14 +544,17 @@ static DWORD WINAPI select_ws_wait_thread(LPVOID lpParameter)
             }
           }
           else {
-            /* if the pipe has been closed, sleep and continue waiting */
+            /* if the pipe has NOT been closed, sleep and continue waiting */
             length = GetLastError();
-            logmsg("[select_ws_wait_thread] PeekNamedPipe error: %d", length);
-            if(length == ERROR_BROKEN_PIPE) {
+            if(length != ERROR_BROKEN_PIPE) {
```

Comment:
It is confusing to reuse the length variable for this. DWORD gle = GetLastError(); You could end up with an error other than ERROR_BROKEN_PIPE, so I'm not sure if this is a good idea.

Reply:
I have another PR in the pipe to completely redo the variable naming for select_ws, so I would rather keep it for this PR.

Reply:
ok

Reply:
Otherwise of course you have valid point. ;-)